Add a post for NYU Depth V2 in 🤗 Datasets

sayakpaul opened the pull request:
Context: https://github.com/huggingface/datasets/issues/5255#issuecomment-1335249427

Google Doc draft: https://docs.google.com/document/d/1jZIXubDqCgQUVtM5o0MLBrrGu4_LnVJabSOjvH6KdfE/edit?usp=sharing

**Reviewers**: Please comment / modify / suggest the Google Doc directly. Once that is locked, I will convert the doc to markdown and add it to the PR.

sayakpaul commented:
@osanseviero I know we have a template https://github.com/huggingface/blog/blob/main/assets/thumbnail-template.svg for blog thumbnail images, but it doesn't have any recommended process. I am finding it a bit difficult to work on thumbnail images that respect our guidelines. Do you mind sharing one if you have one?

lhoestq commented:
Nice ! Added a few comments :)

sayakpaul commented on nyu-depth-v2.md:
````diff
@@ -0,0 +1,175 @@
+
+The complete data-loading script is available [here](https://huggingface.co/datasets/sayakpaul/nyu_depth_v2/blob/main/nyu_depth_v2.py) and we recommend you refer to it when contributing large datasets. The same principle was also followed for the seminal ImageNet-1k dataset and its data-loading script is available [here](https://huggingface.co/datasets/imagenet-1k/blob/main/imagenet-1k.py). Note  ...
+
+With NYU Depth V2 now on 🤗 Datasets, users can load it with just two lines of code:
+
+```py
+from datasets import load_dataset
+dataset = load_dataset("sayakpaul/nyu_depth_v2")
+```
+
+We're also actively working with the original dataset authors to transfer the dataset to an approrpriate organization on the Hub.
````
@osanseviero @lhoestq this is the part I newly added. Rest of the content is same as the Google Doc.

osanseviero replied:
Maybe let's wait some days before publishing and ideally we can connect the comms of the dataset release with the blog post

sayakpaul commented:
Just finished previewing the blog post on Codespaces.

![Screenshot 2022-12-16 at 1 49 48 PM](https://user-images.githubusercontent.com/22957388/208056218-6c490b15-0263-4eba-807c-3953a4ddd388.png)

Things look fine on my end. 

Let me know if we're good to merge.

lhoestq reviewed (approved):
Thanks !

osanseviero reviewed (approved):
Very cool blog post! 

I find it more about creating a loading script for shared files. I think maybe the name/title of the article should change to reflect more about that rather than the use case, see comments below. 

Is all this content covered in the docs btw? Are there any improvements we could do in the docs now that you wrote this?

sayakpaul commented on nyu-depth-v2.md:
```diff
@@ -0,0 +1,177 @@
+title: Welcoming NYU Depth V2 to 🤗 Datasets
+thumbnail: /blog/assets/nyu-depth-v2/thumbnail.png
+---
+
```
> Is all this content covered in the docs btw? Are there any improvements we could do in the docs now that you wrote this?

Covered very well in https://huggingface.co/docs/datasets/dataset_script.

sayakpaul commented on nyu-depth-v2.md:
```diff
@@ -0,0 +1,177 @@
+
+In the above commands, `--max-size` argument denotes the maximum size of a single shard and `--maxshards` denotes the maximum number of shards to be created. In general, having `max-size` as 3e9 (3 GBs) is not a strict requirement and you can keep it to 1e9 (1 GB). You might have to tune the `maxshards` then as well. When selecting the values for `max-size` or `maxshards` a general rule of thumb  ...
+
+You can also check out the Golang port called [`tarp`](https://github.com/webdataset/tarp), but its barrier to entry might seem a little higher.
```
Reworded to reflect that `tarp` is faster in terms of processing time.

sayakpaul commented on nyu-depth-v2.md:
```diff
@@ -0,0 +1,177 @@
+
+The idea would be to shard the single big archive into multiple archives and have multiple workers operate on them individually. Fortunately, 🤗 Datasets does the heavy lifting for us - we only need to take care of the first part – sharding the big archive file into multiple archives. The parallelization part is already taken care of by 🤗 Datasets.
+
+This is what we did for the NYU Depth V2 dataset and achieved a great speedup in the dataset loading time. But how did we shard the TAR archive into multiple ones? Read on to know more.
```
I don't have the exact number, unfortunately.

sayakpaul commented on nyu-depth-v2.md:
```diff
@@ -0,0 +1,177 @@
+
+As with other things at Hugging Face, 🤗 Datasets’ Hub is very much a community-driven effort. Contributing a new dataset is fairly straightforward, and the [official guide](https://huggingface.co/docs/datasets/share) will get you started in no time. But when it comes to contributing larger datasets, things might need additional work.
+
+So, in this post, we’ll share some best practices for working with large datasets by discussing a case study of the [NYU Depth V2 dataset](https://cs.nyu.edu/~silberman/datasets/nyu_depth_v2.html) that we recently added to 🤗 Datasets. The approaches shared in this post are modality agnostic, meaning you can apply the same principles to datasets of other modalities (such as video, text, and audio) ...
```
The primary objective is to make the readers aware of the practices for adding large dataset, especially for computer vision. However, the practices are very much generalizable to other modalities too. But there would be differences in terms of feature decoding, compression format based on the modality. 

WDYT of "Adding a Large Vision Dataset to 🤗 Dataset"?

lhoestq replied:
I also like "How to share a Large Image Dataset ?"

sayakpaul replied:
Don't you think indicating 🤗 Dataset is important?

osanseviero replied:
I think the :hugs: is not very important. The blog shows how to share a large image dataset indeed :fire:

sayakpaul replied:
"Sharing a Large Vision Dataset to 🤗 Datasets" sounds better to me.

osanseviero replied:
Maybe just "Sharing a Large Vision Datasets"? Shorter names are better, and not sure saying "to :hugs: Datasets" is important, as the goal of the article is showing how to share your datasets with others

sayakpaul commented:
@osanseviero addressed your comments. Question for you here: https://github.com/huggingface/blog/pull/718#discussion_r1051788443

Regarding having the figure assets in https://huggingface.co/datasets/huggingface/documentation-images, opened a PR: https://huggingface.co/datasets/huggingface/documentation-images/discussions/17. Once the PR is merged, I will update the links. 

Let me know if there's anything I missed out on.

osanseviero commented:
@sayakpaul I merged the docs PR. Let's hold for some days so @nateraw takes a look and we get the NYU org with the dataset.

sayakpaul commented:
@osanseviero okay. 

Please be informed that @nateraw has already given the [G-Doc version of this](https://docs.google.com/document/d/1jZIXubDqCgQUVtM5o0MLBrrGu4_LnVJabSOjvH6KdfE/edit?usp=sharing) a go.

sayakpaul commented:
@osanseviero I changed the blog title and the URL. 

If they look good, I will change the asset folder names in the respective places. 

Cc: @lhoestq

osanseviero commented on _tags.yml:
```diff
@@ -33,3 +33,10 @@
 - value: diffusion
   label: Diffusion
+
+- value: depth-estimation
```
These are the tags that are shown in hf.co/blog, so we try to have few of them. Would it be possible to remove them? (it's still possible to have them in the `blog.yml` file, so just changes in this file are needed

sayakpaul replied:
I think this is a need tag. Would help with filtering and search.

sayakpaul replied:
Oh now I got your point. Adjusted accordingly.

osanseviero reviewed (approved):
Cool! Left a comment on the tags and the title but otherwise SGTM

sayakpaul commented:
>  Maybe just "Sharing a Large Vision Datasets"? Shorter names are better, and not sure saying "to 🤗 Datasets" is important, as the goal of the article is showing how to share your datasets with others

Agreed but the goal is to show that with 🤗 Datasets and not with any other dataset processing library. So, I believe having that distinction reflected in is necessary. It's beneficial to 🤗 Datasets users as they can immediately be sure of 🤗 Datasets' specific features (such as library library independence and so on).

julien-c commented on _blog.yml:
```diff
@@ -1570,3 +1570,13 @@
   tags:
     - ethics
+
+- local: share-vision-ds
```
```suggestion
- local: sharing-vision-datasets
```

make `ds` more explicit

julien-c commented on share-vision-ds.md:
```diff
@@ -0,0 +1,184 @@
+
+Like other things at Hugging Face, the Datasets’ Hub is a collaborative community-driven space. Contributing a new dataset is fairly straightforward, and the [official guide](https://huggingface.co/docs/datasets/share) will get you started in no time. When it comes to contributing larger datasets, things might need additional work.
+
+So, in this post, we’ll share some best practices for working with large datasets by discussing a case study of the [NYU Depth V2 dataset](https://cs.nyu.edu/~silberman/datasets/nyu_depth_v2.html) that we recently added to 🤗 Datasets. The approaches shared in this post are modality agnostic, meaning you can apply the same principles to datasets of other modalities (such as video, text, and audio) ...
```
i don't think we need to add the `🤗` emoji each time.

makes reading a bit hard/cluttered visually

(if you've seen this in another doc, i would suggest to change the guideline there too :) )

sayakpaul replied:
Do you suggest writing Hugging Face Datasets in place of 🤗 Datasets? 

> (if you've seen this in another doc, i would suggest to change the guideline there too :) )

This then needs to get spread out as a general message within the teams I guess.

osanseviero replied:
When you talk about the OS library, people usually do :hugs: Datasets (same as transformers and accelerate). When people talk about the Hub, we don't put either emoji nor Hugging Face. (except in the first mention in a blog post, there we write "Hugging Face Hub" the first time and "Hub" afterwards)

sayakpaul replied:
Yes, this ^

This is what I followed.

julien-c commented on share-vision-ds.md:
```diff
@@ -0,0 +1,184 @@
+
+## Conclusion
+
+In this post, we shared some practices to handle the contribution of large datasets to 🤗 Datasets. We hope these will come in handy for your future datasets. We cannot wait to see them 🤗
```
maybe cool to link to a few other cool large vision datasets that are on the Hub already

(to act as some kind of landing page for `large vision datasets`)

julien-c reviewed (commented):
left a few nits, but looks cool!
